### ISSUE TITLE
Invoices: can't create clean total price incl.VAT.

### ISSUE THREAD

**Issue by EZS-JD** [salesagility/SuiteCRM#1776]
#### Issue

In some instances there seems to be no way to enter the line item price in such a way that the total incl.VAT is precisely a well-rounded amount (N'NNN.00).

e.g. we have a VAT of 8%. I try to enter a line item price that would result in total price of 2500.00.

Entering 2'314.82 results in 2'500.01
Entering 2'314.81 results in 2'499.99

This is a problem.

The easiest (albeit hidden) workaround would be to not round the list item price to 2 decimals before saving to the database, but rather 3 decimals (as the prices are saved in database with 6 decimals anyhow). Then I could enter and save the list item price as 2'314.815 and the total would result in 2'500.00.
#### Your Environment

- SuiteCRM Version used: 7.4.1
- Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Version 51.0.2704.103 (64-bit)
- Environment name and version (e.g. MySQL, PHP 7): MySQL, PHP7

**Comment by EZS-JD** [comment]
I feel AOS should be set up as flexible as possible to allow for as many use cases as possible. IMHO that would entail:

**1)** AOS allow definition of unit prices with high degree of accuracy (large number of significant figures)
**2)** AOS allows fragments of unit quantities with high degree of accuracy, i.e. high number of decimals (large number of significant figures)

Point **1)** is already pretty much taken care of on the database level with 6 decimals for all price/amount fields (field type "decimal(26,6)"). What’s missing is that this high degree of accuracy be reflected and retained on the UI. JS should not be allowed to strip list prices of their significant figures, ever. That's an unnecessary destruction of information.

=> Remove the discrepancy between what’s written in the database and what’s shown to the user for product prices / line item prices. Make it transparent to the user what the basis of line-item calculations and rounding is.
=> Only allow JS to round to 2 decimals on the resulting amounts (line item totals, line item tax amounts, sums, etc.), not on list prices. (continue to hide trailing "0")

Point **2)** This seems to already be a feature of AOS that was inadvertently removed, but was recently fixed. I believe it is very important that unit quantity fractions are possible (let's say I'm selling currency e.g.). see https://github.com/salesagility/SuiteCRM/issues/2238

With 1) and 2) set up accordingly, AOS would allow for a wide variety of use cases and would also automatically remove the problem originally brought up in this issue, where it's not possible to set list item prices in a way that result in clean summed total amounts.

Nice-to-have: Ability to set/manipulate maximal accuracy of list prices and quantity fractions via admin panel.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Invoices: can't create clean total price incl.VAT.](https://trello.com/c/YtzglZpe/84-invoices-can-t-create-clean-total-price-incl-vat)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
